### PR TITLE
[PRP-273] Save doc upload s3 links to the database

### PR DESCRIPTION
**Pull request by rocketnova**
## Ticket

https://wicmtdp.atlassian.net/browse/PRP-273

- Save s3 presigned url into the `documents` table as a new `s3Url` column
- Add `DOC_URL_EXPIRATION` env var to control s3 presigned url expiration time

This PR saves the presigned url for accepted uploaded documents into the database.

### CODE REVIEW

File: infra/app/envs/dev/main.tf [hunk 1]
```diff
@@ -52,13 +52,20 @@ module "project_config" {
...
-  staff_url             = "${local.environment_name}-staff.wic-services.org"
-  analytics_url         = "${local.environment_name}-analytics.wic-services.org"
+
+  # Urls
+  participant_url = "${local.environment_name}.wic-services.org"
+  staff_url       = "${local.environment_name}-staff.wic-services.org"
+  analytics_url   = "${local.environment_name}-analytics.wic-services.org"
+
+  # Misc settings
+  participant_doc_upload_expiration = "60"
```
Comment (microwavenby) [inline infra/app/envs/dev/main.tf]: .. maybe a little longer than 60 seconds? It's gonna be freaky to see thumbnails randomly disappear in dev..

File: participant/prisma/migrations/20230420184919_add_s3_url/migration.sql [hunk 2]
```diff
@@ -0,0 +1,8 @@
+/*
+  Warnings:
+
+  - Added the required column `s3_url` to the `documents` table without a default value. This is not possible if the table is not empty.
+
+*/
+-- AlterTable
```
Comment (microwavenby) [inline participant/prisma/migrations/20230420184919_add_s3_url/migration.sql:7]: We're not going to be able to ignore these warnings much longer 😅

**Review by microwavenby — APPROVED** [review]

File: infra/app/env-template/main.tf [hunk 3]
```diff
@@ -83,6 +83,10 @@ module "participant" {
     }
   ]
   container_env_vars = [
+    {
+      name  = "DOC_URL_EXPIRATION",
```
Comment (rocketnova) [inline infra/app/env-template/main.tf]: `S3_PRESIGNED_URL_EXPIRATION` might be better

File: infra/app/envs/dev/main.tf (hunk 1 above)
Comment (rocketnova) [reply infra/app/envs/dev/main.tf]: Good call. I'll bump it up to 5m

File: participant/.env.example [hunk 4]
```diff
@@ -12,3 +12,4 @@ MAX_UPLOAD_SIZE_BYTES=5242880
 MAX_UPLOAD_FILECOUNT=5
 AWS_ACCESS_KEY_ID=e2e_minio
 AWS_SECRET_ACCESS_KEY=incredible_local_secret_phrase
+DOC_URL_EXPIRATION=60
```
Comment (rocketnova) [inline participant/.env.example]: Update here too.

File: participant/app/routes/$localAgency/recertify/upload.tsx [hunk 5]
```diff
@@ -265,7 +265,13 @@ export const action = async ({
     )} from form`
   );
   acceptedDocuments.map(async (acceptedFile) => {
-    await upsertDocument(submissionID, acceptedFile!);
+    if (acceptedFile.key) {
+      // Create a presigned URL with expiration time and save to the database.
+      const url = await getURLFromS3(acceptedFile.key, DOC_URL_EXPIRATION);
```
Comment (rocketnova) [inline participant/app/routes/$localAgency/recertify/upload.tsx]: Change s3.server line to default to the env var

**Comment by rocketnova** [comment]
Add a todo to write a unit test in s3.server.ts to test expiration of presigned urls. check it works, sleep 5s, check it fails.

`import { BUCKET, ENDPOINT_URL } from "app/utils/config.server";`

File: participant/app/routes/$localAgency/recertify/upload.tsx [hunk 6]
```diff
@@ -265,7 +265,13 @@ export const action = async ({
     )} from form`
   );
   acceptedDocuments.map(async (acceptedFile) => {
-    await upsertDocument(submissionID, acceptedFile!);
+    if (acceptedFile.key) {
```
Comment (rocketnova) [inline participant/app/routes/$localAgency/recertify/upload.tsx]: Change this to use `acceptedFile!`

**Pull request merged**